test/evp_extra_test.c

```diff
@@ -2571,66 +2573,1012 @@ static int test_EVP_SM2(void)
 
 #endif
 
+#ifndef OPENSSL_NO_ML_KEM
```

Comment:
I dont think we should not keep on stuffing PQ related tests into this file.
It should just be done in a separate ml_kem_test.c file.

Reply:
I strongly agree. How would you structure these? On for all three ML-KEM variants, another for ML-DSA?
Or share some common bits related to the RNG?
Perhaps there's also a better way to stage the RNG entropy injection?
Am I using the right malleable RNG for the test?
This is where I am looking for expertise from you and Pauli, who've been through this with FIPS.

Reply:
See below :).. I dont think there will be much sharing. I tend to just data drive the tests, so in that case a single file for ML_KEM would be used.

Reply:
And a separate one for ML_DSA.

Reply:
Will relocate the tests in a follow-on PR.